lblod/frontend-worship-decisions
# [DL-6071] Add a generic error and 404 page

This adds an error page that will be displayed when an exception is thrown in one of the model hooks. The error page also knows how to deal with 404 API errors and in that case will display a "not found" message.

The actual "not found" page now also displays the same message, so users will see this instead of a white page if they make a typo in the URL.

The 404 page design is based on the one from `vlaanderen.be`.

The error page is a little bare-bones at the moment, but it will do until we have a design for it.

<details>
<summary>Screenshot 404 page</summary>

![Screen Shot 2024-07-31 at 13 48 50](https://github.com/user-attachments/assets/583c85ef-57f1-4f72-bf8f-2c74c24eec28)

</details>

**To test**

1. Log in as a mock user
2. go to the details page of a submission and copy the link
3. log out, and log back in as a different mock user that shouldn't be able to see this submission
4. visit the link you copied
5. it should show the 404 page (after a while, it retries several times before erroring out)

To test the generic page you would need to throw an error in one of the model hooks.
